src/java/org/apache/cassandra/utils/FastByteOperations.java

```diff
@@ -269,16 +272,31 @@ public void copy(ByteBuffer srcBuf, int srcPosition, ByteBuffer trgBuf, int trgP
         {
             Object src;
             long srcOffset;
-            if (srcBuf.hasArray())
+
+            // Heap ByteBuffer (Mutable)
+            if (srcBuf.hasArray() && !srcBuf.isReadOnly())
+            {
+                src = theUnsafe.getObject(srcBuf, HEAP_HB_FIELD);
```

Comment:
rather than rely on the new logic, can we keep the old logic that has been safe for years?  unsafe can break on us in jdk bumps so if we can avoid it in the 99.999% case that is best